File: Lessons/Lesson 01/BinaryGap.kt
// you can also use imports, for example:
// import kotlin.math.*

// you can write to stdout for debugging purposes, e.g.
// println("this is a debug message")

fun solution(N: Int): Int {
    val binary = N.toString(radix = 2)
    var counter = 0
    var gap = 0
    for (digit in binary) {
        if (digit == '0') counter++ else {
            if (counter > gap) gap = counter
            counter = 0
        }
    }
    return gap
}

File: Lessons/Lesson 01/BinaryGap.swift
import Foundation
import Glibc

// you can write to stdout for debugging purposes, e.g.
// print("this is a debug message")

public func solution(_ N : Int) -> Int {
    let binary = String(N, radix: 2)
    var counter = 0
    var gap = 0
    for digit in binary {
        if digit == "0" {
            counter += 1
        } else {
            gap = max(gap, counter)
            counter = 0
        }
    }
    return gap
}
